packages/cozy-stack-client/src/CozyStackClient.js

```diff
@@ -169,7 +174,12 @@ class CozyStackClient {
       return await this.fetchJSONWithCurrentToken(method, path, body, options)
     } catch (e) {
       if (errors.EXPIRED_TOKEN.test(e.message)) {
-        const token = await this.refreshToken()
+        let token
+        try {
+          token = await this.refreshToken()
+        } catch (refreshError) {
+          throw e
```

Comment:
why do we have a try/catch if we only throw the received error ? Isn't this the same as not having any try/catch ?

Reply:
We do not throw the error, we throw the original error (token-expired error), not the tried-to-refresh-error. I think it is clearer for the user since she is not supposed to know that we tried to refresh the token.

Reply:
Oh, you're perfectly right!
I didn't see that.